design/upf-selection-design.md

```diff
@@ -0,0 +1,128 @@
+      - "001010000000008"
+      - "001010000000009"
+    ip-domain-name: "pool1"
+    ip-domain-expanded:
+      dnn: internet
+      dns-primary: "10.176.0.11"        # Value sent to UE
+      mtu: 1460                        # Value sent to UE when PDU Session Established
+      ue-ip-pool: "172.250.1.0/16"     # IP address pool for subscribers
+      ue-dnn-qos:
+        dnn-mbr-downlink: 1000         # UE level downlink QoS (Maximum bit rate per UE)
+        dnn-mbr-uplink: 1000           # UE level uplink QoS (Maximum bit rate per UE)
+        bitrate-unit: Mbps             # Unit for above QoS rates
+        traffic-class:                 # Default bearer QCI/ARP (not used in 5G)
+          name: "platinum"
+          qci: 9
+          arp: 6
+          pdb: 300
+          pelr: 6
+    site-info: "enterprise"
+  - name: "5g-gnbsim-user-group2"
+    imsis:
+      - "001010000000001"
+      - "001010000000002"
+      - "001010000000003"
+      - "001010000000005"
+      - "001010000000008"
+      - "001010000000009"
+    ip-domain-name: "pool1"
+    ip-domain-expanded:
+      dnn: sdcore
+      dns-primary: "10.176.0.11"        # Value sent to UE
+      mtu: 1460                        # Value sent to UE when PDU Session Established
+      ue-ip-pool: "172.250.1.0/16"     # IP address pool for subscribers
+      ue-dnn-qos:
+        dnn-mbr-downlink: 1000         # UE level downlink QoS (Maximum bit rate per UE)
+        dnn-mbr-uplink: 1000           # UE level uplink QoS (Maximum bit rate per UE)
+        bitrate-unit: Mbps             # Unit for above QoS rates
+        traffic-class:                 # Default bearer QCI/ARP (not used in 5G)
+          name: "platinum"
+          qci: 9
+          arp: 6
+          pdb: 300
+          pelr: 6
+
+    **Intended Behavior:** The new design aims to enhance the SMF logic to support UPF selection based on both DNN and Slice ID. This improvement will enable the SMF to correctly handle multiple DNNs within the same Slice ID, ensuring accurate UPF selection and preventing issues like the current failure with the `sdcore` DNN.
+
+## 3. Proposed Changes
+
+### 3.1. SMF Changes
+
+- **Objectives:**
+  - Refactor the SMF to ensure that the UPF selection process accurately considers both the DNN and Slice ID parameters.
+
+- **Action Steps:**
+  - **Code Identification:** Locate the exact portion of the SMF codebase where the UPF selection logic is implemented.
+  - **Logic Enhancement:** Modify the existing logic to incorporate DNN and Slice ID as key criteria in UPF selection.
+  - **PFCP Session Updates:** Update the SMF's PFCP session handling logic to include DNN information, ensuring compatibility with the new selection criteria.
+
+- **Expected Outcome:** By making these changes, the SMF will be able to accurately select the appropriate UPF based on the DNN and Slice ID, resolving issues like the one observed with the sdcore DNN.
+
+### 3.2. UPF Changes
+- **PFCP Association:**
```

Comment:
Let's add the details which needs change. e.g. PFCP association now should send list of DNNs supported. and then

Reply:
Added